app/main/models/models.py

```diff
@@ -114,6 +114,13 @@ class Cluster(BaseModel):
     def __repr__(self):
         return f'<Cluster {self.id}: {self.name}>'
 
+    def count_items(self, count):
+        items = len(self.child_items)
+        count += items
+        for child_cluster in self.child_clusters:
+            count += child_cluster.count_items(count)
+        return count
+
```

Comment:
I'm pretty sure you can get rid of that count argument (and the current version might actually overcount):
```python
def count_items(self):
        count = len(self.child_items)
        for child_cluster in self.child_clusters:
            count += child_cluster.count_items()
        return count
```